openwisp_controller/config/static/config/js/device.js

```diff
@@ -0,0 +1,37 @@
+"use strict";
+django.jQuery(function ($) {
+  if ($(".add-form").length || !$("#device_form").length) {
+    return;
+  }
+
+  // replaces the management ip field with text with the option to edit it
+  var ip_input = $(".field-management_ip > div > input");
+  var initial_ip = ip_input.prop("value");
+  ip_input.after(function () {
+    ip_input.hide();
+    return (
+      '<span class="readonly" id="management_ip_text">' +
+      (initial_ip === "" ? "-n/a-" : initial_ip) +
+      "</span>"
+    );
+  });
+  $("#management_ip_text").after(
+    '<a id="edit_management_ip">  <img src="/static/admin/img/icon-changelink.svg" alt="Edit"></a>'
+  );
```

Comment:
You can remove the whitespace here.

The edit icon looks too close to the text than rest of the edit icons. I checked CSS property of other edit icons. Following properties should make it appear similar:

```
margin-left: 7px;
opacity: 0.8;
```

Please confirm if they works well for you too.